File: ReactMeals_WebApi/Models/Dish.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace ReactMeals_WebApi.Models;

public class Dish
{
    public Dish() { }
    public Dish(int dishId, string dishName, string dishDescription, decimal price)
    {
        DishId = dishId;
        Dish_name = dishName;
        Dish_description = dishDescription;
        Price = price;
    }
    public Dish(int dishId, string dishName, string dishDescription, decimal price, string dishExtendedInfo)
    : this(dishId, dishName, dishDescription, price)
    {
        Dish_extended_info = dishExtendedInfo;
    }
    [Key]
    public int DishId { get; set; }
    public string Dish_name { get; set; }
    public string Dish_description { get; set; }
    [Precision(18, 2)]
    public decimal Price { get; set; }
    public string Dish_extended_info { get; set; }
    public string Dish_url { get; set; }
}

public class DishWithCounter : Dish
{
    public DishWithCounter() { }
    public DishWithCounter(int dishId, string dishName, string dishDescription, decimal price, int dishCounter)
        : base(dishId, dishName, dishDescription, price)
    {
        Dish_counter = dishCounter;
    }
    public int Dish_counter { get; set; }
}

File: src/main/java/gr/jimmys/jimmysfoodzilla/models/Dish.java
package gr.jimmys.jimmysfoodzilla.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import gr.jimmys.jimmysfoodzilla.dto.AddDishDTO;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.Objects;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
@Table(name = "Dishes")
public class Dish {
    @JsonProperty("dishId")
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "DishId")
    private int id;

    @JsonProperty("dish_name")
    @Column(name = " Dish_name", columnDefinition = "VARCHAR(MAX)")
    private String name;

    @JsonProperty("dish_description")
    @Column(name = " Dish_description", columnDefinition = "VARCHAR(MAX)")
    private String description;

    @JsonProperty("price")
    @Column(name = "Price", precision = 18, scale = 2)
    private BigDecimal price;

    @JsonProperty("dish_extended_info")
    @Column(name = " Dish_extended_info", columnDefinition = "VARCHAR(MAX)")
    private String extendedInfo;

    @JsonProperty("dish_url")
    @Column(name = " Dish_url", columnDefinition = "VARCHAR(MAX)")
    private String url;

    @Override
    public boolean equals(Object o) {
        if (o == this)
            return true;
        if (!(o instanceof Dish d))
            return false;
        return this.id == d.getId();
    }

    @Override
    public int hashCode() {
        return this.id; //why not? ID -> distinct value..
    }

    public boolean equalsExceptId(AddDishDTO dto) {
        return Objects.equals(name, dto.getDishName()) &&
                Objects.equals(url, dto.getDishImageBase64()) &&
                Objects.equals(description, dto.getDishDescription()) &&
                Objects.equals(extendedInfo, dto.getDishExtendedInfo());
    }
}

